### PR TITLE
Add press-key-to-login to device auth flow

### PR DESCRIPTION
**Pull request by amcaplan**
### WHY are these changes introduced?

Recently we switched to device auth by default, which has added some friction to login. This restores the simplicity of pressing any key to log in.

### WHAT is this pull request doing?

Before:

Screenshot 2024-07-17 at 16 51 24

After:

Screenshot 2024-07-17 at 16 50 32

We only use this flow in non-cloud TTY environments.

### How to test your changes?

Logout and do anything that requires login, e.g. `config link`

### Measuring impact

How do we know this change was effective? Please choose one:

- [ ] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix
- [x] Existing analytics will cater for this addition
- [ ] PR includes analytics changes to measure impact

### Checklist

- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [x] I've considered possible [documentation](https://shopify.dev) changes

### CODE REVIEW

File: packages/cli-kit/src/private/node/session/device-authorization.ts [hunk 1]
```diff
@@ -47,11 +50,25 @@ export async function requestDeviceAuthorization(scopes: string[]): Promise<Devi
...
+    outputInfo(
+      outputContent`Opened link to start the auth process: ${outputToken.link(
+        'Login link',
+        jsonResult.verification_uri_complete,
+      )}`,
+    )
+  } else {
+    outputInfo(outputContent`User verification code: ${jsonResult.user_code}`)
+    outputInfo(
+      outputContent`👉 Open this link to start the auth process: ${outputToken.link(
```
Comment (shauns) [inline packages/cli-kit/src/private/node/session/device-authorization.ts:66]: I think in this case the `green` approach is better, as we've already established this terminal might not be TTY aware anyway, and it reads a bit better with out the extra formatting we get with `link` in unsupported environments

Comment (amcaplan) [reply packages/cli-kit/src/private/node/session/device-authorization.ts:66]: Is that actually true? In cloud environments, are links generally non-clickable?

**Review by shauns — APPROVED** [review]
Looking good 👍

**Comment by phrfpeixoto** [comment]
This change broke my development process using docker containers. Please see #5496

**Pull request merged**